alerting/src/main/kotlin/org/opensearch/alerting/alertsv2/AlertV2Mover.kt

```diff
@@ -0,0 +1,439 @@
+             the IDs of the old triggers. this means Alerting will generate a new ID for the updated triggers by default,
+             meaning this logic will pick up those updated triggers and correctly move/delete the alerts
+            */
+            if (monitorV2 != null) {
+                boolQuery.mustNot(QueryBuilders.termsQuery(AlertV2.TRIGGER_V2_ID_FIELD, monitorV2.triggers.map { it.id }))
+            }
+
+            val alertsSearchQuery = SearchSourceBuilder.searchSource()
+                .query(boolQuery)
+                .size(10000)
+                .version(true)
+            val activeAlertsRequest = SearchRequest(ALERT_V2_INDEX)
+                .source(alertsSearchQuery)
+            val searchAlertsResponse: SearchResponse = client.suspendUntil { search(activeAlertsRequest, it) }
+
+            // If no alerts are found, simply return
+            if (searchAlertsResponse.hits.totalHits?.value == 0L) return
+
+            val activeAlerts = mutableListOf<AlertV2>()
+            searchAlertsResponse.hits.forEach { hit ->
+                activeAlerts.add(
+                    AlertV2.parse(
+                        XContentHelper.createParser(
+                            NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE,
+                            hit.sourceRef, XContentType.JSON
+                        ),
+                        hit.id,
+                        hit.version
+                    )
+                )
+            }
+
+            // pre-index the alerts so retrieving their
+            // monitor IDs for routing is easier
+            val alertsById: Map<String, AlertV2> = activeAlerts.associateBy { it.id }
+
+            val alertV2HistoryEnabled = monitorCtx.clusterService!!.clusterSettings.get(ALERT_V2_HISTORY_ENABLED)
+
+            // if alert v2 history is enabled, migrate the relevant alerts
+            // to the alert v2 history index pattern instead of hard deleting them
+            var copyResponse: BulkResponse? = null
+            if (alertV2HistoryEnabled) {
+                val indexRequests = searchAlertsResponse.hits.map { hit ->
+                    val xcp = XContentHelper.createParser(
+                        NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE,
+                        hit.sourceRef, XContentType.JSON
+                    )
+
+                    IndexRequest(ALERT_V2_HISTORY_WRITE_INDEX)
+                        .routing(monitorV2Id)
+                        .source(
+                            AlertV2.parse(xcp, hit.id, hit.version)
+                                .toXContentWithUser(XContentFactory.jsonBuilder())
+                        )
+                        .version(hit.version)
+                        .versionType(VersionType.EXTERNAL_GTE)
+                        .id(hit.id)
+                }
+                val copyRequest = BulkRequest().add(indexRequests)
```

Comment:
Do we need to consider batching the write/delete calls? I'm wondering what would happen in situations when a customer could possibly have hundreds of alerts from multiple monitors that need managing. Or is the retry logic below expected to handle that kind of situation?